pyvista/core/filters/poly_data.py

```diff
@@ -438,12 +438,8 @@ def merge(
         )
 
         # convert back to a polydata if both inputs were polydata
-        if is_pd:
-            pd_merged = pyvista.PolyData(merged.points, faces=merged.cells, n_faces=merged.n_cells)
-            pd_merged.point_data.update(merged.point_data)
-            pd_merged.cell_data.update(merged.cell_data)
-            pd_merged.field_data.update(merged.field_data)
-            merged = pd_merged
+        if is_polydata:
+            merged = merged.extract_geometry()
```

Comment:
Maybe add a comment about the reason we do this and the potential performance hit?